Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -380,21 +380,24 @@ static bool isRendererThreadSafe(WebSampleBufferVideoRendering *renderering)
     resetReadyForMoreSample();
 }
 
-void VideoMediaSampleRenderer::decodedFrameAvailable(RetainPtr<CMSampleBufferRef>&& sample)
+VideoMediaSampleRenderer::DecodedFrameResult VideoMediaSampleRenderer::decodedFrameAvailable(RetainPtr<CMSampleBufferRef>&& sample)
 {
     assertIsCurrent(dispatcher());
 
     assignResourceOwner(sample.get());
 
-    if (maybeQueueFrameForDisplay(sample.get()))
+    DecodedFrameResult result = maybeQueueFrameForDisplay(sample.get());
+    if (result == DecodedFrameResult::TooLate)
         m_droppedVideoFrames++;
     else
         PAL::CMBufferQueueEnqueue(m_decodedSampleQueue.get(), sample.get());
 
     purgeDecodedSampleQueue();
+
+    return result;
```

Comment:
Why not returning a boolean telling whether to call `notifyHasAvailableVideoFrame`?

Reply:
> Why not returning a boolean telling whether to call `notifyHasAvailableVideoFrame`?

I also like that it documents what happened. So I could remove the comment.